source/sync/data-access-patterns/flexible-sync-permissions-guide.txt

```diff
@@ -0,0 +1,607 @@
+  to enable Sync with these permissions.
+
+
+.. _write-own-data-read-all-data:
+
+Write Own Data, Read All Data
+-----------------------------
+
+In this case, users can read all data, but write only their own data. Consider a 
+recipe app where users can read all recipes and add new recipes. The recipes 
+they add can be viewed by everyone using the app. 
+
+To set this up, follow these steps:
+
+.. include:: /includes/use-flex-sync-templates
+
+- :guilabel:`Define Permissions`: use the template dropdown to
+  select the template called "Users can read all data but only write their own data".
+
+  This populates the rule expression box with the following:
+
+  .. code-block:: json
+     :emphasize-lines: 6-7
+  
+     {
+        "defaultRoles": [
+          {
+            "name": "owner-write", 
+            "applyWhen": {},
+            "read": true,
+            "write": {"owner_id": "%%user.id"}
+          }    
+        ]  
+      }
+
+  Note that the "read" expression is set to ``true``, indicating that no matter
+  which user is authenticated, they can access all of the data. The "write"
+  expression uses the ``owner_id`` field we marked as "queryable" above and uses
+  the ``%%user`` "expansion" to match against the requesting user's id. Realm
+  replaces (or "expands") the expansion at evaluation time with the
+  corresponding value -- in this case, the user object.
+
+  When a user accesses the data through Sync, Realm evaluates these
+  expressions to determine permissions. This expression says the requesting user
+  may read all data, but write only if the document's ``owner_id`` field is equal 
+  to the user's ID.
+
+- Skip :guilabel:`Advanced Configuration`, and then click :guilabel:`Save Changes` 
+  to enable Sync with these permissions.
+
+.. _admin-privileges:
+
+Administrator Privileges
+------------------------
+
+In this permission strategy, users with a specific "administrator" role can read
+and write any document. Users who do not have the specified role can only read
+and write their own data. To make this strategy work, you first need to define
+which users have administrator permissions.
+
+MongoDB Realm has a feature called Custom User Data that allows you to associate
+arbitrary data with each user. Using this feature, you can create a document
+with a field that indicates whether the user has administrative privileges.
+While there are many ways to set this up, one approach is to add a boolean
+property called ``isAdmin``, which is set to ``true`` for those users with the
+elevated permissions. Another is to create a string field called ``role``, in
+which one of the expected values might be "admin". 
+
+In the following example, the custom user object we'll create has an ``_id``
+field, which corresponds to the user's ID, and 3 additional fields:
+``firstName``, ``lastName``, and ``isGlobalAdmin``:
+
+.. code-block:: json
+
+   {
+      "_id" : "1234",
```

Comment:
Aren't user ids in realm ObjectIDs? Might be good to align this with the actual product implementation if so. Otherwise, feel free to ignore this.

Reply:
Nope. User IDs in Stitch/Realm/App Services are strings.